[FIX] website: Displays all dependent fields

kevin11andrew opened the pull request:
Description of the issue/feature this PR addresses:
Previously, when we had more than one dependent field on checkboxes, only the first dependent field that met its requirement was displayed. Minor edits were made to go through each instance in the form to see if the requirement for a given dependent field is met and all of the fields for which the requirements are met are displayed.

Current behavior before PR:
When we had more than one dependent field on check-boxes, only the first dependent field that met its requirement was displayed. 

Desired behavior after PR is merged:
All dependent fields for which the requirement is met is displayed.



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

R0-00 reviewed (changes requested):
Let us not break the current behavior of "Is equal to". What we could actually maybe do is make the (already existing for other fields) "contains" option appear for multiple checkbox.


Also for the commit messages
1. No need of uppercase to start your commit title..
2. Your commit message should not overflow 72 char per line. You could use multiple lines to write the commit message.

kevin11andrew commented:
I made the changes that you specified

R0-00 reviewed (changes requested):
The "Is equal to" and "Is not equal to" is not working properly now.
Also can't see "contains" on the visibility condition selection for multiple checkbox. 


Regarding you commit.
There should only be one commit. You can squash all other commits.

kevin11andrew commented:
Can you check it now

R0-00 commented on addons/website/static/src/snippets/s_website_form/000.js:
```diff
@@ -635,10 +641,11 @@ odoo.define('website.s_website_form', function (require) {
                 const formData = new FormData(this.$target[0]);
-                const currentValueOfDependency = formData.get(dependencyName);
-                return this._compareTo(comparator, currentValueOfDependency, visibilityCondition, between);
+                if(comparator==='contains' || comparator==='!contains')
```
You can replace the double check by includes so you don't have to write twice comparator and should we add a third check it will be easier: ["contains", "!contains"].includes(comparator)

R0-00 reviewed (changes requested):
The code works, but you have made changes inside the _compareTo.

 You can actually  use the old implementation of _compareTo itself. All you have to do is make changes to an input of _compareTo. 
 
This will reduce the number of code changes.



Regarding commit message
- You can change the commit message since you have not made changes to the previous behavior and have implemented contains for the checkboxes.
- In the commit message just give a small description about the issue and how you fixed it.

kevin11andrew reviewed (commented):
Made the changes you specified

R0-00 commented:
> Made the changes you specified

There are a few more changes that I had requested

kevin11andrew commented:
I altered the fix in the way u specified and it seems to work as well.

R0-00 reviewed (changes requested):
Now the old functionality of "is equal to" is not working on the multiple checkbox. 

(For example if I have 3 checkboxes and 3 fields dependent on them using "is equal to" (visible when equal to), when I select 2 or more checkboxes, none of the fields are visible.)





Regarding commit message

* You can change the commit message since you are not making changes to the previous behavior and have implemented contains for the multiple checkboxes.
* In the commit message just give a small description about the issue and how you fixed it.

kevin11andrew committed:
[FIX] website: displays all dependent fields
Previously, when we had more than one dependent field on checkboxes,
only the first dependent field that met its requirement was displayed.
Minor edits were made to go through each instance in the form to see
if the requirement for a given dependent field is met and all of the
fields for which the requirements are met are displayed. Moreover,
the current functionality of 'is equal to' has not been altered.
Instead the contians field handles the above requirement.

R0-00 reviewed (approved):
Your code is good now. No need for further changes.
